integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/search/LuceneMultiValueSearchPredicateIT1.java

```diff
@@ -0,0 +1,927 @@
+ */
+@RunWith(Parameterized.class)
+public class LuceneMultiValueSearchPredicateIT1 {
+
+	private static final boolean SKIP_TEST_DOUBLE = false;
+	private static final boolean SKIP_TEST_FLOAT = false;
+	private static final boolean SKIP_TEST_LONG = false;
+	private static final boolean SKIP_TEST_INTEGER = false;
+
+	private static final boolean SKIP_TEST_PARENT = false;
+	private static final boolean SKIP_TEST_FLATTENDED = false;
+	private static final boolean SKIP_TEST_NESTED = false;
+
+	private final String VALUES = "data/search/multivalues-data.json";
+
+	// Backend 1 / Index 1
+	private static final String INDEX_NAME_1_1 = "IndexName_1_1";
+
+	private static final String DOCUMENT_1_1_1 = "1_1_1";
+
+	private static final String DOCUMENT_1_1_2 = "1_1_2";
+
+	// Backend 1 / Index 2
+	private static final String INDEX_NAME_1_2 = "IndexName_1_2";
+
+	private static final String DOCUMENT_1_2_1 = "1_2_1";
+	private static final String DOCUMENT_1_2_2 = "1_2_2";
+
+	private final String directoryType;
+
+	@Rule
+	public SearchSetupHelper setupHelper = new SearchSetupHelper();
+
+	// Backend 1 / Index 1
+	private IndexMapping_1_1 indexMapping_1_1;
+	private StubMappingIndexManager indexManager_1_1;
+
+	// Backend 1 / Index 2
+	private IndexMapping_1_2 indexMapping_1_2;
+	private StubMappingIndexManager indexManager_1_2;
+
+	@Parameterized.Parameters(name = "Lucene directory type {0}")
+	public static Object[] data() {
+		return new Object[]{"local-heap", "local-filesystem"};
+	}
+
+	public LuceneMultiValueSearchPredicateIT1(String directoryType) {
+		this.directoryType = directoryType;
+	}
+
+	@Before
+	public void setup() {
+		setupHelper.start()
+			.withBackendProperty( "directory.type", directoryType )
+			.withIndex(
+				INDEX_NAME_1_1,
+				ctx -> this.indexMapping_1_1 = new IndexMapping_1_1( ctx.getSchemaElement() ),
+				indexMapping -> this.indexManager_1_1 = indexMapping
+			)
+			.withIndex(
+				INDEX_NAME_1_2,
+				ctx -> this.indexMapping_1_2 = new IndexMapping_1_2( ctx.getSchemaElement() ),
+				indexMapping -> this.indexManager_1_2 = indexMapping
+			)
+			.setup();
+
+		initData();
+	}
+
+	@Test
+	@TestForIssue(jiraKey = "HSEARCH-3839")
+	public void double_searchParentMultivalues() {
+
+		if ( SKIP_TEST_PARENT ) {
+			return;
+		}
+
+		if ( SKIP_TEST_DOUBLE ) {
+			return;
+		}
+
+		StubMappingScope scope = indexManager_1_1.createScope();
+
+		SearchQuery<DocumentReference> query = scope.query()
+			.where( f -> f.matchAll() )
+			.sort( f -> f.field( "additionalDoubleField" ).asc().multi().min() )
+			.toQuery();
+
+		assertThat( query ).hasDocRefHitsExactOrder( c -> {
+			c.doc( INDEX_NAME_1_1, DOCUMENT_1_1_2, DOCUMENT_1_1_1 );
+		} );
+
+		query = scope.query()
+			.where( f -> f.matchAll() )
+			.sort( f -> f.field( "additionalDoubleField" ).asc().multi().max() )
+			.toQuery();
+
+		assertThat( query ).hasDocRefHitsExactOrder( c -> {
+			c.doc( INDEX_NAME_1_1, DOCUMENT_1_1_2, DOCUMENT_1_1_1 );
```

Comment:
I know the data sample is too small. But whatever happens in the code there will be an exception rather than a test done differently. The test checks the behavior inside, tests all possible modes of work inside the code. There are currently four types of variables (Double, Float, Long, Integer). Including three possible ways to perform these functions (Master, Flattended, Nested). In each of these cases, different code is executed.